### PR TITLE
add -DNOMINMAX to both WIN32 and WIN64

### PR DESCRIPTION
**Pull request by cynthiajoan**
***

***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [x] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: firestore/CMakeLists.txt [hunk 1]
```diff
@@ -347,10 +347,11 @@ add_custom_target(
 # are guarded by this flag, such as GetUserAgent and function_registry.
 set(FIREBASE_FIRESTORE_CPP_DEFINES -DINTERNAL_EXPERIMENTAL=1)
 
-if (WIN32 AND NOT ANDROID AND NOT IOS)
+if (MSVC AND NOT ANDROID AND NOT IOS)
   # On Windows, gRPC gives a compiler error in firebase_metadata_provider_desktop.cc
   # unless _WIN32_WINNT is defined to this value (0x0600, Windows Vista).
   # Also set -DNOMINMAX for both Firestore and Firestore Core.
+  message("SET -DNOMINMAX for firestore cpp build")
```
Comment (dconeybe) [inline firestore/CMakeLists.txt:354]: Can this message be removed? It appears to only have served the debugging of the cmake build configuration.

File: firestore/CMakeLists.txt [hunk 2]
```diff
@@ -362,6 +363,7 @@ if (WIN32 AND NOT ANDROID AND NOT IOS)
     _LIBCPP_VERSION=99)
 endif()
 
+message("FIREBASE_FIRESTORE_CPP_DEFINES is ${FIREBASE_FIRESTORE_CPP_DEFINES}")
```
Comment (dconeybe) [inline firestore/CMakeLists.txt:366]: Can this message be removed as well?

File: firestore/src/include/firebase/firestore/field_value.h [hunk 3]
```diff
@@ -376,6 +376,7 @@ class FieldValue final {
   static FieldValue Increment(T by_value) {
     // Note: Doxygen will run into trouble if this function's definition is
     // moved outside the class body.
+    #undef max
```
Comment (dconeybe) [inline firestore/src/include/firebase/firestore/field_value.h:379]: Is this `#undef max` still required?

**Review by dconeybe — CHANGES_REQUESTED** [review]
Thanks for fixing this! I've left some drive-by comments. Do you know why this hasn't been a problem in the 64-bit Windows C++ SDK builds but is in the Unity builds?

**Comment by dconeybe** [comment]
Please ignore my comments. I did not pay close attention to the purpose of this PR.

File: firestore/src/include/firebase/firestore/field_value.h (hunk 3 above)
Comment (cynthiajoan) [reply firestore/src/include/firebase/firestore/field_value.h]: removed

File: firestore/CMakeLists.txt (hunk 1 above)
Comment (cynthiajoan) [reply firestore/CMakeLists.txt]: removed

**Review by dconeybe — APPROVED** [review]

**Pull request merged**